### DISCUSSION TITLE
Video file names cut shorter

### DISCUSSION THREAD

**Discussion by braille8011** [aclap-dev/video-downloadhelper#1239] [Help and Q&A] [answered]
### Your question

Firefox extension, Version 9.1.0.42, Companion App v.2.0.19.

When downloading videos, the file names are being shortened to exactly 64 characters in length. The destination folder is only named D:\Downloaded Videos, so the names aren't being shortened to accomodate the Windows file name length limit. From my research, Firefox itself doesn't have any setting to shorten the file names of downloaded files, so I have to assume this extension is doing this itself, even though I can't find any setting in the options panel that would set this.

Any help would be greatly appreciated!

### Addon & CoApp version and details

version: 9.1.0.42
target: mozilla
channel: stable
lang: en-US
coapp: {"found":true,"path":"C:\\Program Files\\DownloadHelper CoApp\\vdhcoapp.exe","version":"2.0.19","new_version":false}
license: {"accepted":true,"email":"<removed>","key":"53a609e6f7054d7b"}
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:130.0) Gecko/20100101 Firefox/130.0
{
    "valid_license_message_has_been_displayed": true,
    "used_history_button": true,
    "download_directory": "D:\\Downloaded Videos",
    "never_show_no_incognito_msg_again": true,
    "theme": "system",
    "view_options": {
        "all_tabs": false,
        "low_quality": false,
        "sort_by_status": true,
        "sort_reverse": false,
        "show_button_clean": true,
        "show_button_clean_all": false,
        "show_button_convert_local": false,
        "hide_downloaded": true
    },
    "use_sidebar": true,
    "open_count_store": 2,
    "successfull_dl": 21,
    "never_show_successfull_dl_message": true
}

**Comment by GrampaWildWilly** [comment]
I do not work for VDH.  I'm just another user.

It is completely pathological & not just non-intuitive but anti-intuitive.  Take a look at #838.  There are several screenshots in there.  Much of the discussion will probably not interest you.  But the setting for the length of a file name is in there.

**Comment by braille8011** [comment, marked as answer]
I found the answer. In the three dots menu for the download is an option for Smartnaming, which opens an options tab that shows the default is set to 64 characters max for the file name. This shows it was the app doing it, not sure why this option is so low or hard to find. I set it to 192 and saved it for all hosts, so it shouldn't be an issue now. Posting the solution here in case anyone finds this post looking for the same thing.

**Comment by as-muncher** [comment]
@GrampaWildWilly Thanks for the help.
@paulrouget  Intuitively, one would click the video download helper icon (I'm using firefox), and then look through the settings to change the number of characters by default.  Really, people should not have to look on github or have to post something to search so hard for this setting.  When you're downloading a video, you don't want to have to choose the default number of characters right there; you want a setting that you can set regardless of whether there are any videos at hand.  So, please change this so that the setting can be changed easily.  64 as a minimum is too low, in my opinion, because I often see filenames cut short.  128 I think should be the default.  Now, I have to go try to find a video to download so that I can change that stupid default setting.  Yeah, I should not have to search for a video, then select Smartnaming, then select 128 and then click on "set for all hostnames".

**Reply by GrampaWildWilly** [reply]
Yeah, 64 is too low as a default.  On the other hand, it does have the benefit --  maybe "benefit" -- of forcing people to find the Smartnaming dialog.

**Reply by as-muncher** [reply]
No, not a benefit, because people get upset.  It forces people to have to find the Smartnaming dialogue, forces people to have to look through github maybe, maybe get through it, or perhaps just using a different downloader.

**Reply by braille8011** [reply]
Yeah, I don't see any benefit here. 1) the option should be part of the standard settings that are easy to find, not a hidden menu that only appears when a download is found by the extension, and 2) the default should be "max" and be accompanied by an explanation that, at this setting, the file names will be truncated according to "Windows max character length - download folder character length", and show the math, which would be something like "255 - 10 = 245 max characters".

**Reply by GrampaWildWilly** [reply]
I mentioned this to @paulrouget Paul a while ago, that Smartnaming needs to have some of its options on the regular VDH `More settings` menu, & it should not be subordinate to finding some content that VDH can recognize.  He said it was a good idea & he would put it on his to-do list.  I don't believe he has shared his to-do list with anybody, certainly not with me.  So your guess is as good as mine when he will get around to fixing this.

**Reply by GrampaWildWilly** [reply]
By the way @braille8011, the solution you marked as the answer here is exactly what you would have learned if you had read the reference I gave you.  It would have no doubt saved you some trial & error if you had clicked that link & read what you found there.

**Reply by braille8011** [reply]
I appreciate that your link had the answer I was looking for, but please note that I I didn't see your post until shortly after I found the answer on my own when I went here to post about it. It didn't come up in any search I tried on this site, either. So, at the end of it all, with how difficult it would be for a new user to find the answer to this question, we're both just proving my point that this issue is far too difficult to resolve for anyone less than a power user who loves this extension and doesn't want to switch to another over an issue like this.
